### PR TITLE
Change lexical scope msp

### PR DESCRIPTION
**Pull request by haslinghuis**

### CODE REVIEW

File: src/js/msp/MSPHelper.js [hunk 1]
```diff
@@ -2256,12 +2260,14 @@ MspHelper.prototype.crunch = function(code) {
 
         case MSPCodes.MSP_MULTIPLE_MSP:
 
-            while (FC.MULTIPLE_MSP.msp_commands.length > 0) {
-
-                let mspCommand = FC.MULTIPLE_MSP.msp_commands.shift();
-
-                self.mspMultipleCache.push(mspCommand);
-                buffer.push8(mspCommand);
+            while (true) {
```
Comment (mikeller) [inline src/js/msp/MSPHelper.js:2263]: Confusing - please remove.

File: src/js/msp/MSPHelper.js [hunk 2]
```diff
@@ -2579,33 +2585,34 @@ MspHelper.prototype.sendLedStripConfig = function(onCompleteCallback) {
...
     function send_next_led_strip_config() {
 
-        var led = FC.LED_STRIP[ledIndex];
-        var ledDirectionLetters =        ['n', 'e', 's', 'w', 'u', 'd'];      // in LSB bit order
-        var ledFunctionLetters =         ['i', 'w', 'f', 'a', 't', 'r', 'c', 'g', 's', 'b', 'l']; // in LSB bit order
-        var ledBaseFunctionLetters =     ['c', 'f', 'a', 'l', 's', 'g', 'r']; // in LSB bit
+        const buffer = [];
+
+        const led = FC.LED_STRIP[ledIndex];
+        const ledDirectionLetters =        ['n', 'e', 's', 'w', 'u', 'd'];      // in LSB bit order
```
Comment (mikeller) [inline src/js/msp/MSPHelper.js:2591]: This is actually repeated from https://github.com/betaflight/betaflight-configurator/pull/2308/files#diff-2a7a24b74c7564918c438782f9a53a977205c0c79fd016a069b6dca546563adfR1202 - should be de-duplicated.

File: src/js/msp/MSPHelper.js [hunk 3]
```diff
@@ -2710,13 +2713,13 @@ MspHelper.prototype.sendLedStripModeColors = function(onCompleteCallback) {
...
-    var functions = [];
+    const self = this;
+    const functions = [];
 
-    var keys = Object.keys(self.SERIAL_PORT_FUNCTIONS);
-    for (var index = 0; index < keys.length; index++) {
-        var key = keys[index];
-        var bit = self.SERIAL_PORT_FUNCTIONS[key];
+    const keys = Object.keys(self.SERIAL_PORT_FUNCTIONS);
+    for (const index of keys) {
```
Comment (mikeller) [inline src/js/msp/MSPHelper.js:2720]: `for ( of )` returns a value and not an index.

File: src/js/msp/MSPHelper.js [hunk 4]
```diff
@@ -2725,13 +2728,12 @@ MspHelper.prototype.serialPortFunctionMaskToFunctions = function(functionMask) {
...
-    var self = this;
-    var mask = 0;
+    const self = this;
+    let mask = 0;
 
-    var keys = Object.keys(self.SERIAL_PORT_FUNCTIONS);
-    for (var index = 0; index < functions.length; index++) {
-        var key = functions[index];
-        var bitIndex = self.SERIAL_PORT_FUNCTIONS[key];
+    for (const index of functions) {
```
Comment (mikeller) [inline src/js/msp/MSPHelper.js:2734]: `for ( of )` returns a value and not an index.

**Review by mikeller — CHANGES_REQUESTED** [review]

**Comment by haslinghuis** [comment]
Thanks. All Fixed as requested per review

**Comment by haslinghuis** [comment]
Sonar is very confusing about this. Changed back to old style.
image

File: src/js/msp/MSPHelper.js [hunk 5]
```diff
@@ -2725,13 +2715,12 @@ MspHelper.prototype.serialPortFunctionMaskToFunctions = function(functionMask) {
...
-    var self = this;
-    var mask = 0;
+    const self = this;
+    let mask = 0;
 
-    var keys = Object.keys(self.SERIAL_PORT_FUNCTIONS);
-    for (var index = 0; index < functions.length; index++) {
-        var key = functions[index];
-        var bitIndex = self.SERIAL_PORT_FUNCTIONS[key];
+    for (const index in functions) {
```
Comment (mikeller) [inline src/js/msp/MSPHelper.js:2721]: One more time: It is incorrect to replace `for ( ;; )` with `for ( in )`. Please stop doing this.

**Review by mikeller — CHANGES_REQUESTED** [review]

**Review by mikeller — APPROVED** [review]

**Pull request merged**